src/main/webapp/app/exercises/shared/import/from-file/exercise-import-from-file.component.ts

```diff
@@ -0,0 +1,102 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { Exercise, ExerciseType } from 'app/entities/exercise.model';
+import { MAX_FILE_SIZE } from 'app/shared/constants/input.constants';
+import { AlertService } from 'app/core/util/alert.service';
+import { faUpload } from '@fortawesome/free-solid-svg-icons';
+import { ProgrammingExercise, ProgrammingLanguage, ProjectType } from 'app/entities/programming-exercise.model';
+import JSZip from 'jszip';
+
+@Component({
+    selector: 'jhi-exercise-import-from-file',
+    templateUrl: './exercise-import-from-file.component.html',
+})
+export class ExerciseImportFromFileComponent implements OnInit {
+    @Input()
+    exerciseType: ExerciseType;
+    titleKey: string;
+    fileForImport?: File;
+    faUpload = faUpload;
+    @Input()
+    exercise: Exercise;
```

Comment:
Can you sort this into constants for the html, input and other variables?

Reply:
done